src/consensus/consensus.h

```diff
@@ -6,10 +6,36 @@
 #ifndef BITCOIN_CONSENSUS_CONSENSUS_H
 #define BITCOIN_CONSENSUS_CONSENSUS_H
 
+#include <stdint.h>
+
+static const uint64_t BCIP103_FORK_TIME = 1456790400; // April 01 2016, midnight UTC
+
 /** The maximum allowed size for a serialized block, in bytes (network rule) */
-static const unsigned int MAX_BLOCK_SIZE = 1000000;
+inline unsigned int MaxBlockSize(uint64_t nTime) {
+    if (nTime < BCIP01_FORK_TIME)
+        return 1000*1000;
+
+    // cap for tests
+    if (nTime > 4113158400)
+        nTime = 4113158400;
+
+/** Smooth transition of 30 bytes every 10 minutes, until 3 megabytes is reached */
+    if (((2*1000*1000) + (30 * ((nTime - BCIP01_FORK_TIME) / 600))) <= 3000000) {
+      return (2*1000*1000) + (30 * ((nTime - BCIP01_FORK_TIME) / 600));
```

Comment:
Segregated Witness might be helpful here
Do we have a decision on https://github.com/bitcoinclassic/bitcoinclassic/pull/8 yet?